ngx_http_sxg_filter_module.c

```diff
@@ -267,18 +276,28 @@ static bool calc_integrity(const ngx_http_request_t* const req,
 
 static ngx_int_t subresource_fetch_handler(ngx_http_request_t* req, void* data,
                                            ngx_int_t rc) {
-  ngx_http_sxg_ctx_t* ctx = data;
-  ngx_http_core_srv_conf_t* cscf =
-      ngx_http_get_module_srv_conf(req, ngx_http_core_module);
   if (req->done) {
     return NGX_OK;
   }
-  if (req->upstream->headers_in.status_n != 200) {
+  ngx_http_core_srv_conf_t* cscf =
+      ngx_http_get_module_srv_conf(req, ngx_http_core_module);
+  ngx_http_sxg_ctx_t* ctx = data;
+
+  pthread_mutex_lock(&ctx->lock);
```

Comment:
Please check the return codes throughout.

Reply:
OK, I appended return code check for mutex API.